File: src/main/java/compiler/parser/operator_parsers/AddParser.java
package compiler.parser.operator_parsers;

import compiler.block.Block;
import compiler.block.operators.Add;
import compiler.parser.Parser;
import compiler.tokenizer.Tokenizer;

public class AddParser extends Parser<Add> {
    @Override
    public boolean shouldParse(String line) {
        return line.matches("[a-zA-Z][a-zA-Z0-9]* [+][=] [0-9]+;");
    }

    @Override
    public Add parse(Block superBlock, Tokenizer tokenizer) {

        String name = tokenizer.nextToken().getToken();

        tokenizer.nextToken().getToken();

        String value = tokenizer.nextToken().getToken();

        return new Add(superBlock,name,value);
    }
}

File: src/main/scala/compiler/parser/operators/AddParser.scala
package compiler.parser.operators

import compiler.block.Block
import compiler.block.operators.AddBlock
import compiler.parser.Parser
import compiler.tokenizer.Tokenizer
import test_classes.block.Block
import test_classes.block.operators.AddBlock
import test_classes.parser.Parser
import test_classes.tokenizer.Tokenizer

class AddParser extends Parser[AddBlock] {
  def shouldParse(line: String): Boolean = line.matches("[a-zA-Z][a-zA-Z0-9]*[ ]*[+][=][ ]*[0-9]+")

  def parse(superBlock: Block, tokenizer: Tokenizer): AddBlock = {
    val name: String = tokenizer.nextToken.token
    tokenizer.nextToken
    val value: String = tokenizer.nextToken.token
    return new AddBlock(superBlock, name, value)
  }
}
